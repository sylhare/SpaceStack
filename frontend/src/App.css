body {
  background-color: #92a8d1;
  padding: 20px;
  margin: 0;
}

h1,
h2,
h3,
p,
span,
label,
ul,
i,
li {
  font-family: Helvetica, Arial, sans-serif;
  line-height: 1.8;
}

ul.header li {
  display: inline;
  list-style-type: none;
  margin: 0;
}

ul.header li:hover a:hover {
  background-color: #2b2b2b;
}

ul.header {
  background-color: #111;
  padding: 0;
  border-radius: 4px;
}

ul.header li a {
  color: #FFF;
  font-weight: bold;
  text-decoration: none;
  padding: 20px;
  display: inline-block;
}

.content {
  background-color: #FFF;
  border-radius: 4px;
  padding: 20px;
}

.content h2 {
  padding: 0;
  margin: 0;
}

.content li {
  margin-bottom: 10px;
}

.active {
  background-color: #034f84;
}

/* Style inputs, select elements and textareas */
input[type=text], select, textarea {
  flex: 0 1 auto;
  padding: 12px;
  border: 1px solid #ccc;
  border-radius: 4px;
  box-sizing: border-box;
  resize: vertical;
}

/* Style the label to display next to the inputs */
label {
  padding: 12px 12px 12px 0;
  display: inline-block;
}

/* Style the submit button */
input[type=submit] {
  flex: 0 1 auto;
  background-color: #f7786b;
  color: white;
  padding: 12px 12px;
  border: none;
  border-radius: 4px;
  cursor: pointer;
  margin-left: 10px;
}

button {
  flex: 0 1 auto;
  color: white;
  padding: 6px 6px;
  border: none;
  border-radius: 4px;
  cursor: pointer;
  margin-right: 10px;
}

button.approved {
  background-color: #77a8a8;
}

button.denied {
  background-color: #c94c4c;
}

input[type=submit]:hover {
  background-color: #f7cac9;
}

input[type=submit]:disabled {
  background: #dddddd;
  pointer-events: none;
}

input[type=submit]:focus {
  background-color: #f7786b;
}


form#post-request-form {
  display: flex;
  flex-direction: column;
}

#post-request-form p {
  flex: 1 1 auto;
}

ul.baby-requests {
  list-style-type: none;
  margin: 0;
  padding: 0;
}

.baby-requests span {
  padding-right: 2px;
  padding-left: 2px;
  white-space: nowrap;
}

span.baby-name {
  margin-left: 10px;
}

.PopulationCount p, h3 {
  display: inline-block;
}

.PopulationCount h3 {
  margin-left: 10px;
}

.error {
  color: #db2269;
  font-size: 0.7em;
  display: block;
  margin-left: 2px;
}

.hidden {
  display: none;
  visibility: hidden;
}

.snackbar-container {
  display: flex;
  width: 95%;
  align-items: center;
  justify-content: center;
  bottom: 30px;
  z-index: 1;
  position: fixed;
}

.snackbar {
  visibility: hidden;
  font-family: "Segoe UI", "Ubuntu", "Roboto", "Open Sans", "Helvetica Neue", sans-serif;
  min-width: 250px;
  background-color: #2b2b2b;
  color: #fff;
  text-align: center;
  border-radius: 4px;
  padding: 15px 30px;
  font-size: 1em;
}

.snackbar.show {
  visibility: visible;
  -webkit-animation: fadein 0.5s, fadeout 0.5s 2.5s;
  animation: fadein 0.5s, fadeout 0.5s 2.5s;
}

@-webkit-keyframes fadein {
  from {bottom: 0; opacity: 0;}
  to {bottom: 30px; opacity: 1;}
}

@keyframes fadein {
  from {bottom: 0; opacity: 0;}
  to {bottom: 30px; opacity: 1;}
}

@-webkit-keyframes fadeout {
  from {bottom: 30px; opacity: 1;}
  to {bottom: 0; opacity: 0;}
}

@keyframes fadeout {
  from {bottom: 30px; opacity: 1;}
  to {bottom: 0; opacity: 0;}
}

.processed-requests {
  border-collapse: collapse;
  margin: 25px 0;
  font-size: 0.9em;
  font-family: sans-serif;
  min-width: 400px;
  box-shadow: 10px 10px 50px -5px rgba(0, 0, 0, 0.15);
}

.processed-requests thead tr {
  background-color: #f7786b;
  color: #ffffff;
  text-align: left;
}

.processed-requests th,
.processed-requests td {
  padding: 12px 15px;
}

.processed-requests tbody tr {
  border-bottom: 1px solid #dddddd;
}

.processed-requests tbody tr:nth-of-type(odd) {
  background-color: #ffffff;
}

.processed-requests tbody tr:nth-of-type(even) {
  background-color: #f3f3f3;
}

.processed-requests tbody tr:last-of-type {
  border-bottom: 2px solid #f7786b;
}

.processed-requests tbody tr.active-row {
  font-weight: bold;
  color: #f7786b;
}

.login {
  display: flex;
  width: 95%;
  align-items: center;
  justify-content: center;
  margin: 30px 0;
  font-size: 1.2em;
}

label.login-input {
  padding-bottom: 5px;
  margin-top: 20px;
}

form {
  width: 50%;
}

input[type=text].login-input, input[type=password].login-input {
  width: 100%;
  padding: 12px 20px;
  margin: 8px 0;
  display: inline-block;
  border: 1px solid #ccc;
  box-sizing: border-box;
  border-radius: 4px;
}

.flex-container {
  display: flex;
  align-items: center;
  justify-content: center;
}

button.login-button {
  background-color: #f7786b;
  color: white;
  padding: 14px 50px;
  margin: 20px 0;
  border: none;
  cursor: pointer;
  max-width: 50%;
  float: right;
}

button:hover {
  opacity: 0.8;
}

.container {
  padding: 16px;
}

span.psw {
  float: right;
  padding-top: 16px;
}

li.logout {
  float: right;
}

input[type="text"].filter {
  width: 62em;
  background-image: url('Resources/searchicon.png');
  padding: 12px 20px 12px 40px;
  background-position: 10px 10px;
  background-repeat: no-repeat;
}
