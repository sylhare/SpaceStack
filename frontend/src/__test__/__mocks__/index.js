import configureStore from "redux-mock-store";

export const mockStore = configureStore([]);

export const loggedInStore = mockStore({
  authReducer: {
    isLoggedIn: true,
  }
});

export const loggedOutStore = mockStore({
  authReducer: {
    isLoggedIn: false,
  }
});
