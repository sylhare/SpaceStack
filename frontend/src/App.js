import React from 'react';
import {BrowserRouter as Router, NavLink, Redirect, Route} from 'react-router-dom';
import './App.css';

import Pioneers from './Pages/Pioneers';
import Manage from './Pages/Manage';
import Home from './Pages/Home';
import Audit from './Pages/Audit';
import Login from './Pages/Login';
import {useDispatch, useSelector} from "react-redux";
import {logout} from "./Actions/auth";

function App() {
  const isLoggedIn = useSelector(state => state.authReducer.isLoggedIn);
  const dispatch = useDispatch();
  const logOut = () => {
    dispatch(logout());
  };


  return (
    <Router>
      <div>
        <Navbar isLoggedIn={isLoggedIn} logOut={logOut}/>

        <Route path='/login' data-test='login-route' exact component={Login}/>
        <PrivateRoute isLoggedIn={isLoggedIn} path='/' exact component={Home}/>
        <PrivateRoute isLoggedIn={isLoggedIn} path='/pioneers/' component={Pioneers}/>
        <PrivateRoute isLoggedIn={isLoggedIn} path='/manage/' component={Manage}/>
        <PrivateRoute isLoggedIn={isLoggedIn} path='/audit/' component={Audit}/>
      </div>
    </Router>
  );
}

const Navbar = ({isLoggedIn, logOut}) =>
  (<nav>
      <ul className='header'>
        {!isLoggedIn && (
          <li><NavLink aria-label='Login' to='/login' exact activeClassName='active'>Login</NavLink></li>)}

        {isLoggedIn && (
          <React.Fragment>
            <li><NavLink aria-label='Home' data-test='Home' to='/' exact activeClassName='active'>Home</NavLink></li>
            <li><NavLink aria-label='Pioneers' to='/pioneers/' exact activeClassName='active'>Pioneers</NavLink></li>
            <li><NavLink aria-label='Manage' to='/manage/' exact activeClassName='active'>Habitat and Survival
              Management</NavLink></li>
            <li><NavLink aria-label='Audit' to='/audit/' exact activeClassName='active'>Audit</NavLink></li>
            <li aria-label='Logout' className='logout'><a href='/' data-test='logout' onClick={logOut}>Logout</a></li>
          </React.Fragment>
        )}
      </ul>
    </nav>
  );

const PrivateRoute = ({component: Component, ...rest}) => (
  <Route {...rest} render={(props) => (rest.isLoggedIn ? <Component {...props} /> : <Redirect to='/login'/>
  )}/>);

export default App;
