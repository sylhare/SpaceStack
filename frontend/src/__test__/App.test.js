import React from 'react'
import App from '../App';
import renderer from 'react-test-renderer';
import {Provider} from 'react-redux';
import {loggedInStore, loggedOutStore} from "./__mocks__";

const renderApp = (store) => (
  renderer.create(
    <Provider store={store}>
      <App/>
    </Provider>
  )
);

test('Renders App', () => {
  let component = renderApp(loggedOutStore);

  expect(component.toJSON()).toMatchSnapshot();
});

test('Only log in page should be displayed when logged out', () => {
  let component = renderApp(loggedOutStore);

  expect(component.root.findByProps({'data-test': 'login-route'})).toBeTruthy();
  expect(component.toJSON()).not.toContain(/pioneers/i)
});

test('Login page does not appear when logged in', () => {
  let component = renderApp(loggedInStore);

  renderer.act(() => {
  });

  expect(component.toJSON()).not.toContain(/login/i);
  expect(component.root.findByProps({'data-test': 'Home'})).toBeTruthy();
});

test('Logout should log out', () => {
  let store = loggedInStore;
  store.dispatch = jest.fn().mockImplementation(() => Promise.resolve());
  let component = renderApp(store);

  renderer.act(() => {
    component.root.findByProps({'data-test': 'logout'}).props.onClick();
  });

  expect(store.dispatch).toHaveBeenCalledTimes(1);
});
