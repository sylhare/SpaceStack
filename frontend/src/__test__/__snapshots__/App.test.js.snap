// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders App 1`] = `
<div>
  <nav>
    <ul
      className="header"
    >
      <li>
        <a
          aria-current="page"
          aria-label="Login"
          className="active"
          href="/login"
          onClick={[Function]}
          style={Object {}}
        >
          Login
        </a>
      </li>
    </ul>
  </nav>
  <div
    className="content"
  >
    <div
      className="login"
    >
      <form
        onSubmit={[Function]}
      >
        <label
          className="login-input"
        >
          <b>
            Username
          </b>
        </label>
        <input
          className="login-input"
          data-test="username"
          onChange={[Function]}
          placeholder="Enter Username"
          type="text"
          value=""
        />
        
        <label
          className="login-input"
        >
          <b>
            Password
          </b>
        </label>
        <input
          className="login-input"
          data-test="password"
          onChange={[Function]}
          placeholder="Enter Password"
          type="password"
          value=""
        />
        
        <div
          className="flex-container"
        >
          <button
            aria-label="login"
            className="login-button"
            data-test="submit"
            type="submit"
          >
            Log in
          </button>
        </div>
      </form>
    </div>
  </div>
</div>
`;
